cobalt/browser/metrics/cobalt_metrics_service_client.h

```diff
@@ -0,0 +1,181 @@
+  // ownership.
+  void OnEnvironmentUpdate(std::string* serialized_environment) override;
+
+  // Called by the metrics service to record a clean shutdown.
+  void OnLogCleanShutdown() override {}
+
+  // Called prior to a metrics log being closed, allowing the client to
+  // collect extra histograms that will go in that log. Asynchronous API -
+  // the client implementation should call |done_callback| when complete.
+  void CollectFinalMetricsForLog(const base::Closure& done_callback) override;
+
+  // Get the URL of the metrics server.
+  std::string GetMetricsServerUrl() override;
+
+  // Get the fallback HTTP URL of the metrics server.
+  std::string GetInsecureMetricsServerUrl() override;
+
+  // Creates a MetricsLogUploader with the specified parameters (see comments on
+  // MetricsLogUploader for details).
+  std::unique_ptr<::metrics::MetricsLogUploader> CreateUploader(
+      base::StringPiece server_url, base::StringPiece insecure_server_url,
+      base::StringPiece mime_type,
+      ::metrics::MetricsLogUploader::MetricServiceType service_type,
+      const ::metrics::MetricsLogUploader::UploadCallback& on_upload_complete)
+      override;
+
+  // Returns the standard interval between upload attempts.
+  base::TimeDelta GetStandardUploadInterval() override;
+
+  // Called on plugin loading errors.
+  void OnPluginLoadingError(const base::FilePath& plugin_path) override {}
+
+  // Called on renderer crashes in some embedders (e.g., those that do not use
+  // //content and thus do not have //content's notification system available
+  // as a mechanism for observing renderer crashes).
+  void OnRendererProcessCrash() override {}
+
+  // Returns whether metrics reporting is managed by policy.
+  bool IsReportingPolicyManaged() override;
+
+  // Gets information about the default value for the metrics reporting checkbox
+  // shown during first-run.
+  ::metrics::EnableMetricsDefault GetMetricsReportingDefaultState() override;
+
+  // Returns whether cellular logic is enabled for metrics reporting.
+  bool IsUMACellularUploadLogicEnabled() override;
+
+  // Returns true iff sync is in a state that allows UKM to be enabled.
+  // See //components/ukm/observers/sync_disable_observer.h for details.
+  bool SyncStateAllowsUkm() override;
+
+  // Returns true iff sync is in a state that allows UKM to capture extensions.
+  // See //components/ukm/observers/sync_disable_observer.h for details.
+  bool SyncStateAllowsExtensionUkm() override;
+
+  // Returns whether UKM notification listeners were attached to all profiles.
+  bool AreNotificationListenersEnabledOnAllProfiles() override;
+
+  // Gets the Chrome package name for Android. Returns empty string for other
+  // platforms.
+  std::string GetAppPackageName() override;
+
+  explicit CobaltMetricsServiceClient(
+      ::metrics::MetricsStateManager* state_manager, PrefService* local_state);
+
+ private:
+  // Any prefs/state specific to Cobalt Metrics.
+  PrefService* local_state_;
```

Comment:
I may be misreading, but seems unused in this class ? In any case, please nullptr-initialize

Reply:
Ah, this was a remnant of a past refactor. Removed. Good cathc.